packages/components/src/internal/components/forms/input/FileInput.tsx

```diff
@@ -104,7 +111,25 @@ class FileInputImpl extends DisableableInput<FileInputImplProps, State> {
             return;
         }
 
-        this.setFormValue(fileList[0]);
+        const file = fileList[0];
+        if (acceptedFormats) {
+            const formatCheck = fileMatchesAcceptedFormat(file.name, acceptedFormats);
+            if(!formatCheck.isMatch) {
+                this.setState({ error: 'Invalid file type.' });
+                return;
+            }
+
+        }
+
+        if (maxFileSize && file.size > maxFileSize) {
+            this.setState({ error: 'File must not exceed acceptable size.' });
```

Comment:
should we include the maxFileSize in the error message here?

Reply:
I'm tempted to do so but I worry about rounding issue when we do division to make the msg human readable...

Reply:
Does the user have any place where they can see the file limit? If this is the only place, I think we'll have to do something to make this message more informative (perhaps the fileSize value is in KB instead of bytes, so we can do the multiplication when checking the size and use the KB number in the message?).

Reply:
Showing msg in KB seems a good compromise. Updated to show KB converted msg.